pkg/virtctl/hotplug/cpu/cpu.go

```diff
@@ -0,0 +1,132 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 The KubeVirt Contributors
+ *
+ */
+
+package cpu
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"github.com/spf13/cobra"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/clientcmd"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+func NewCPUCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	c := &CPU{clientConfig: clientConfig}
+
+	cmd := &cobra.Command{
+		Use:     "cpu (VM) (CPU_COUNT)",
+		Short:   "Set the number of CPUs (sockets) for a running virtual machine instance.",
+		Example: cpuUsage(),
+		Args:    templates.ExactArgs("cpu", 2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return c.Run(cmd, args)
+		},
+	}
+
+	cmd.Flags().DurationVar(&c.timeout, "timeout", 10*time.Second, "Timeout duration (e.g., 10s, 2m)")
+
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	return cmd
+}
+
+type CPU struct {
+	clientConfig clientcmd.ClientConfig
+	timeout      time.Duration
+}
+
+func cpuUsage() string {
+	usage := `  # Set the number of CPUs (sockets) to 2 for VirtualMachine 'myvm':
+  {{ProgramName}} hotplug cpu myvm 2`
+
+	return usage
+}
+
+func (c *CPU) Run(cmd *cobra.Command, args []string) error {
+	namespace, _, err := c.clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+
+	vmName := args[0]
+	cpuCount, err := strconv.Atoi(args[1])
+	if err != nil {
+		return fmt.Errorf("invalid CPU count: %v", err)
+	}
+
+	if cpuCount <= 0 {
+		return fmt.Errorf("cpuCount parameter must be a positive integer")
+	}
+
+	virtCli, err := kubecli.GetKubevirtClientFromClientConfig(c.clientConfig)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+	defer cancel()
+
+	vm, err := virtCli.VirtualMachine(namespace).Get(ctx, vmName, k8smetav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if vm.Spec.Instancetype != nil {
+		return fmt.Errorf("manual modification of the CPU is not allowed when using an instance type")
+	}
+
+	if err := c.patchCpu(ctx, virtCli, vm, uint32(cpuCount)); err != nil {
+		return err
+	}
+
+	cmd.Printf("Successfully set %d CPUs for %s\n", cpuCount, vmName)
+	return nil
+}
+
+func (c *CPU) patchCpu(ctx context.Context, virtCli kubecli.KubevirtClient, vm *v1.VirtualMachine, cpuCount uint32) error {
+	var cpuToPatch *v1.CPU
+	if vm.Spec.Template.Spec.Domain.CPU != nil {
+		cpuToPatch = vm.Spec.Template.Spec.Domain.CPU
+	} else {
+		cpuToPatch = &v1.CPU{}
+	}
```

Comment:
```suggestion
	cpuToPatch := vm.Spec.Template.Spec.Domain.CPU
	if cpuToPatch == nil {
		cpuToPatch = &v1.CPU{}
	}
```

Would that work? Looks simpler IMO

Reply:
Thanks !